aospy/utils/times.py

```diff
@@ -571,3 +503,89 @@ def ensure_time_as_index(ds):
             da[TIME_STR] = ds[TIME_STR]
             ds[name] = da
     return ds
+
+
+def infer_year(date):
+    """Given a datetime-like object or string infer the year
+
+    Parameters
+    ----------
+    date : str
+        Input date
+
+    Returns
+    -------
+    int
+    """
+    if isinstance(date, str):
+        pattern = '(?P<year>\d{4})'
```

Comment:
Could you add a comment to describe in words what this pattern is looking for, and/or an example?